src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -498,9 +498,8 @@ void C2_MacroAssembler::fast_unlock_lightweight(Register obj, Register box, Regi
 
       if (OMCacheHitRate) increment(Address(rthread, JavaThread::unlock_lookup_offset()));
       ldr(t1_monitor, Address(box, BasicLock::object_monitor_cache_offset_in_bytes()));
-      // TODO: Cleanup these constants (with an enum and asserts)
-      cmp(t1_monitor, (uint8_t)2);
-      // Non symmetrical, take slow path monitor == 0 or 1, 0 and 1 < 2, both LS and NE
+      // null check with Flags == NE, no valid pointer below alignof(ObjectMonitor*)
+      cmp(t1_monitor, checked_cast<uint8_t>(alignof(ObjectMonitor*)));
```

Comment:
I don't know what this means.

Reply:
Maybe I should say `valid ObjectMonitor* check` instead of `null check`. 

The reason that the check used to be against `2` is because some iterations of the code have signalled information in the lowest bit. 

Currently it could just be a check agains `1` which would check for `nullptr` while also setting Flags to NE. 

The check is simply that based on the fact that no valid ObjectMonitor* except nullptr can exist below `alignof(ObjectMonitor*)`.